codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -495,20 +518,22 @@ private void incrementPendingBytes(int numBytes) {
              * Note: this does not flush the {@link ChannelHandlerContext}.
              */
             int write(int allowedBytes) {
-                int before = payload.size();
-                needFlush |= payload.write(Math.max(0, allowedBytes));
-                int writtenBytes = before - payload.size();
                 try {
+                    pendingWriteQueue.remove();
```

Comment:
Actually, I think calling `error` only once from the FlowController is not a problem.  And that's the only thing we should be documenting in the interface: the interaction between FlowController and FlowControlled.  This contract doesn't cover the FlowControlled calling itself - that's beyond the scope of this particular API.  We should document that this can happen, but it should be docs in our encoder code rather than in this interface.